docs/interface/boundary.md

```diff
@@ -0,0 +1,44 @@
+# Boundary related classes
```

Comment:
Thanks for adding this. Some documentation on the boundary classes is long overdue.